controller/search.go

```diff
@@ -140,53 +142,133 @@ func (c *SearchController) WorkitemsCSV(ctx *app.WorkitemsCSVSearchContext) erro
+		}
+		wisCSV, _, err = ConvertWorkItemsToCSV(ctx.Context, c.db, wits, wiResultWindow, false)
+		if err != nil {
+			return goa.ErrBadRequest(fmt.Sprintf("error converting work items to output format for expression '%s': %s", *ctx.FilterExpression, err))
+		}
+		ctx.ResponseWriter.Write([]byte(wisCSV))
+		if len(wiResultWindow) < limit {
+			break
+		}
+		offset = offset + len(wiResultWindow)
+	}
+	// finally, return the result
+	return ctx.OK(nil)
+}
+
+// getWorkItemsByFilterExpression retrieves Work Items for a given expression and parameters
+func getWorkItemsByFilterExpression(ctx context.Context, db application.DB, filterExpression string, filterParentexists *bool, offset *int, limit *int) ([]workitem.WorkItem, error) {
+	var result []workitem.WorkItem
+	err := application.Transactional(db, func(appl application.Application) error {
 		var err error
-		result, _, _, _, err = appl.SearchItems().Filter(ctx.Context, *ctx.FilterExpression, ctx.FilterParentexists, &offset, &limit)
+		result, _, _, _, err = appl.SearchItems().Filter(ctx, filterExpression, filterParentexists, offset, limit)
 		if err != nil {
 			log.Error(ctx, map[string]interface{}{
 				"err":               err,
-				"filter_expression": *ctx.FilterExpression,
+				"filter_expression": filterExpression,
 			}, "unable to list the work items")
-			return errs.Wrapf(err, "error executing filter expression for CSV filtering: %s", *ctx.FilterExpression)
+			return errs.Wrapf(err, "error executing filter expression for CSV filtering: %s", filterExpression)
 		}
 		return nil
 	})
-	if err != nil {
-		return err
-	}
-	// Load all work item types
-	wits, err := loadWorkItemTypesFromArr(ctx.Context, c.db, result)
-	if err != nil {
-		return goa.ErrBadRequest(fmt.Sprintf("error loading work items types while searching work items for expression '%s': %s", *ctx.FilterExpression, err))
-	}
-	// Convert them to CSV format the return value contains the final CSV
-	wisCSV, err := ConvertWorkItemsToCSV(ctx.Context, c.db, wits, result)
-	if err != nil {
-		return goa.ErrBadRequest(fmt.Sprintf("error converting work items to output format for expression '%s': %s", *ctx.FilterExpression, err))
-	}
-	if len(result) >= limit {
-		// we have more than limit+1 rows, so add a note to the bottom of the returned list
-		wisCSV = wisCSV + "\nWIT_NOTE_MORE: There are more result entries. You may want to narrow down your query or use paging to retrieve more results."
-	}
-	// to make filename dynamic, hash of the query expression
-	currentTime := time.Now().UTC()
-	// creating timestr according to RFC3339 recommendations
-	timeStr := currentTime.Format(time.RFC3339)
-	// return output
-	ctx.ResponseData.Header().Set("Content-Disposition", "attachment; filename='workitems-"+timeStr+".csv'")
-	return ctx.OK([]byte(wisCSV))
+	return result, err
+}
+
+// getAllWorkItemTypesBySpaceID retrieves all Work Items for a given spaceID
```

Comment:
```suggestion
// getAllWorkItemTypesBySpaceID retrieves all Work Items Types for a given spaceID
```

Reply:
I do not find this very useful because this function wraps just two function calls.

Reply:
Yeah, will reintegrate that, factored it out to focus on the main algorithm.

Reply:
Fixed efad070.